src/store/contact-groups.test.ts

```diff
@@ -0,0 +1,231 @@
+
+			const list1Copy = [...list1];
+			const secondLast = list1.splice(list1.length - 2, 1)[0];
+			addContactGroups(list1);
+			expect(useContactGroupStore.getState().orderedContactGroups).toEqual(list1);
+			useContactGroupStore.getState().addContactGroupInSortedPosition(secondLast);
+			expect(useContactGroupStore.getState().orderedContactGroups).toEqual(list1Copy);
+			expect(useContactGroupStore.getState().offset).toEqual(1);
+		});
+		it('should add the last element in the unordered list and not increment offset', () => {
+			const list1 = times(20, () => buildContactGroup());
+			list1.sort((a, b) => compareContactGroupName(a.title, b.title));
+
+			const last = list1.splice(list1.length - 1, 1)[0];
+			addContactGroups(list1);
+			expect(useContactGroupStore.getState().orderedContactGroups).toEqual(list1);
+			useContactGroupStore.getState().addContactGroupInSortedPosition(last);
+			expect(useContactGroupStore.getState().orderedContactGroups).toEqual(list1);
+			expect(useContactGroupStore.getState().unorderedContactGroups).toEqual([last]);
+			expect(useContactGroupStore.getState().offset).toEqual(0);
+		});
+		it('should add the last secondLast and thirdLast element in the unordered list, sort the unordered list and not increment offset', () => {
+			const list1 = times(20, () => buildContactGroup());
+			list1.sort((a, b) => compareContactGroupName(a.title, b.title));
+
+			const last = list1.splice(list1.length - 1, 1)[0];
+			const secondLast = list1.splice(list1.length - 1, 1)[0];
+			const thirdLast = list1.splice(list1.length - 1, 1)[0];
+			addContactGroups(list1);
+			useContactGroupStore.getState().addContactGroupInSortedPosition(last);
+			useContactGroupStore.getState().addContactGroupInSortedPosition(thirdLast);
+			useContactGroupStore.getState().addContactGroupInSortedPosition(secondLast);
+
+			expect(useContactGroupStore.getState().orderedContactGroups).toEqual(list1);
+			expect(useContactGroupStore.getState().unorderedContactGroups).toEqual([
+				thirdLast,
+				secondLast,
+				last
+			]);
+			expect(useContactGroupStore.getState().offset).toEqual(0);
+		});
+
+		it('should add element with the same name of the last element in the unordered list', () => {
+			const list1 = times(20, () => buildContactGroup());
+			list1.sort((a, b) => compareContactGroupName(a.title, b.title));
+
+			const withSameNameOfLast = buildContactGroup({ title: last(list1)?.title });
+			addContactGroups(list1);
+			expect(useContactGroupStore.getState().orderedContactGroups).toEqual(list1);
+			useContactGroupStore.getState().addContactGroupInSortedPosition(withSameNameOfLast);
+			expect(useContactGroupStore.getState().orderedContactGroups).toEqual(list1);
+			expect(useContactGroupStore.getState().unorderedContactGroups).toEqual([withSameNameOfLast]);
+			expect(useContactGroupStore.getState().offset).toEqual(0);
+		});
+	});
+
+	describe('UpdateContactGroup action', () => {
```

Comment:
There is a missing test: should update position in ordered list when update the display name with a name before the last element name